wifite/util/crack.py

```diff
@@ -79,26 +78,43 @@ def run(cls):
                 dep_list = ', '.join([dep.dependency_name for dep in deps])
                 Color.pl('     {R}* {R}%s {W}({O}%s{W})' % (tool, dep_list))
 
-        Color.p('\n{+} Enter the {C}cracking tool{W} to use ({C}%s{W}): {G}' % (
-            '{W}, {C}'.join(available_tools.keys())))
-        tool_name = raw_input()
-        if tool_name not in available_tools:
-            Color.pl('{!} {R}"%s"{O} tool not found, defaulting to {C}aircrack{W}' % tool_name)
-            tool_name = 'aircrack'
-        elif any_pmkid and tool_name != 'hashcat':
+        if any_pmkid:
             Color.pl('{!} {O}Note: PMKID hashes will be cracked using {C}hashcat{W}')
+            tool_name = 'hashcat'
+        else:
+            Color.p('\n{+} Enter the {C}cracking tool{W} to use ({C}%s{W}): {G}' % (
+                '{W}, {C}'.join(available_tools.keys())))
+            tool_name = raw_input()
+            if tool_name not in available_tools:
+                Color.pl('{!} {R}"%s"{O} tool not found, defaulting to {C}aircrack{W}' % tool_name)
+                tool_name = 'aircrack'
 
         try:
             for hs in hs_to_crack:
                 cls.crack(hs, tool_name)
         except KeyboardInterrupt:
             Color.pl('\n{!} {O}Interrupted{W}')
 
+    @classmethod
+    def is_cracked(cls, file):
+        if not os.path.exists(Configuration.cracked_file):
+            return False
+        with open(Configuration.cracked_file) as f:
+            json = loads(f.read())
+        if json is None:
+            return False
+        for result in json:
+            for k in result.keys():
+                v = result[k]
+                if 'file' in k and v.split('/')[1] == file:
```

Comment:
Instead of splitting on the path separator, Python can extract the filename via [`os.path.basename(v)`](https://docs.python.org/3/library/os.path.html#os.path.basename).

```python
>>> os.path.basename('/abc/def/asdf.txt')
'asdf.txt'
```

Reply:
Agree with you, will be changed.
(by the way thanks for the basename() tip, usually i use os.path.split() )